## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 442939`
Commit: `dce979e4889d94e830c21ce84cf50af1a4779573`

---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>poetryPlugins.poetry-plugin-export</li>
    <li>poetryPlugins.poetry-plugin-export.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-poeblix</li>
    <li>poetryPlugins.poetry-plugin-poeblix.dist</li>
    <li>poetryPlugins.poetry-plugin-shell</li>
    <li>poetryPlugins.poetry-plugin-shell.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>poetryPlugins.poetry-plugin-export</summary>
<pre>        exporter.export(&quot;requirements.txt&quot;, tmp_path, &quot;requirements.txt&quot;)
    
        with (tmp_path / &quot;requirements.txt&quot;).open(encoding=&quot;utf-8&quot;) as f:
            content = f.read()
    
&gt;       assert content == &quot;\n&quot;.join(expected) + &quot;\n&quot;
E       assert &#x27;bar==1.2.2 ;...ion &lt; &quot;4.0&quot;\n&#x27; == &#x27;bar==1.2.2 ;...ion &lt; &quot;4.0&quot;\n&#x27;
E         
E         Skipping 81 identical leading characters in diff, use -v to show
E           n &lt; &quot;4.0&quot;
E         + baz==1.2.3 ; python_version == &quot;2.7&quot; or python_version &gt;= &quot;3.6&quot; and python_version &lt; &quot;4.0&quot;
E           foo==1.2.1 ; python_version == &quot;2.7&quot; or python_version &gt;= &quot;3.6&quot; and python_version &lt; &quot;4.0&quot;

tests/test_exporter.py:1882: AssertionError
------------------------------ Captured log setup ------------------------------
WARNING  poetry.core.factory:factory.py:64 The &quot;poetry.dev-dependencies&quot; section is deprecated and will be removed in a future version. Use &quot;poetry.group.dev.dependencies&quot; instead.
=========================== short test summary info ============================
FAILED tests/test_exporter.py::test_exporter_exports_requirements_txt_without_dev_packages_by_default[1.1] - assert &#x27;bar==4.5.6 ;...ha256:12345\n&#x27; == &#x27;foo==1.2.3 ;...ha256:12345\n&#x27;
FAILED tests/test_exporter.py::test_exporter_exports_requirements_txt_with_dev_extras[1.1-False-expected1] - assert &#x27;bar==1.2.2 ;...ion &lt; &quot;4.0&quot;\n&#x27; == &#x27;bar==1.2.2 ;...ion &lt; &quot;4.0&quot;\n&#x27;
======================== 2 failed, 139 passed in 5.53s =========================</pre>
</details>
<details>
<summary>poetryPlugins.poetry-plugin-up</summary>
<pre>E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {&#x27;dependency-groups&#x27;: {&#x27;dev&#x27;: [&#x27;fred==1.1.1&#x27;, &#x27;plugh (&gt;=2.2.2,&lt;3.0.0)&#x27;, &#x27;xyzzy (&gt;=2.2.2,&lt;3.0.0)&#x27;, &#x27;nacho (&gt;=2.2.2,&lt;3.0...: [&#x27;foobar (&gt;=2.2.2,&lt;3.0.0)&#x27;
E         
E         ...Full output truncated (2 lines hidden), use &#x27;-vv&#x27; to show

tests/e2e/test_e2e.py:386: AssertionError
------------------------------ Captured log setup ------------------------------
WARNING  poetry.core.packages.path_dependency:path_dependency.py:83 Path /absolute/path/to/my-package/dist/garply-0.1.0.tar.gz for garply does not exist
------------------------------ Captured log call -------------------------------
WARNING  virtualenv.seed.embed.base_embed:base_embed.py:38 The --no-wheel and --wheel options are deprecated. They have no effect for Python &gt; 3.8 as wheel is no longer bundled in virtualenv.
WARNING  poetry.core.packages.path_dependency:path_dependency.py:83 Path /absolute/path/to/my-package/dist/garply-0.1.0.tar.gz for garply does not exist
WARNING  poetry.core.packages.path_dependency:path_dependency.py:83 Path /absolute/path/to/my-package/dist/garply-0.1.0.tar.gz for garply does not exist
WARNING  poetry.core.packages.path_dependency:path_dependency.py:83 Path /absolute/path/to/my-package/dist/garply-0.1.0.tar.gz for garply does not exist
WARNING  poetry.core.packages.path_dependency:path_dependency.py:83 Path /absolute/path/to/my-package/dist/garply-0.1.0.tar.gz for garply does not exist
WARNING  poetry.core.packages.path_dependency:path_dependency.py:83 Path /absolute/path/to/my-package/dist/garply-0.1.0.tar.gz for garply does not exist
=========================== short test summary info ============================
FAILED tests/e2e/test_e2e.py::test_command_with_latest_project - AssertionError: assert {&#x27;project&#x27;: {....2,&lt;3.0.0)&#x27;]}} == {&#x27;project&#x27;: {...1....
FAILED tests/e2e/test_e2e.py::test_command_preserve_wildcard_project - AssertionError: assert {&#x27;project&#x27;: {....2,&lt;3.0.0)&#x27;]}} == {&#x27;project&#x27;: {...1....
=================== 2 failed, 50 passed, 1 skipped in 11.56s ===================</pre>
</details>
</details>

--- a/pkgs/by-name/po/poetry/plugins/poetry-plugin-export.nix
+++ b/pkgs/by-name/po/poetry/plugins/poetry-plugin-export.nix
@@ -2,7 +2,6 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  fetchpatch,
   poetry,
   poetry-core,
   pytest-mock,
@@ -12,27 +11,15 @@
 
 buildPythonPackage rec {
   pname = "poetry-plugin-export";
-  version = "1.9.0";
-  format = "pyproject";
+  version = "1.9.0-unstable-2025-09-14";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "python-poetry";
     repo = "poetry-plugin-export";
-    tag = version;
-    hash = "sha256-AP3/njzbLEi2s4pOUSLLLzqNprvxwLe9LSY7qh08EWc=";
+    rev = "70a2f386a52687adee7353b51e59dd45aa319ee7";
+    hash = "sha256-KsvkM4hjG+jrdPVauXYdc6E87Gp7srMg/mJHpWRjaEs=";
   };
-
-  patches = [
-    # Remove after next release of poetry-plugin-export
-    (fetchpatch {
-      url = "https://github.com/python-poetry/poetry-plugin-export/commit/16637f194e86708913ec6e09064c713eb0715bb6.patch";
-      includes = [
-        "tests/test_exporter.py"
-        "tests/markers.py"
-      ];
-      hash = "sha256-ncz9kqp18+yeRXlhmLEcWfO1bDavjohhmVw6DwTy1hA=";
-    })
-  ];
 
   build-system = [
     poetry-core
@@ -48,10 +35,10 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
-  meta = with lib; {
+  meta = {
     changelog = "https://github.com/python-poetry/poetry-plugin-export/blob/${src.rev}/CHANGELOG.md";
     description = "Poetry plugin to export the dependencies to various formats";
-    license = licenses.mit;
+    license = lib.licenses.mit;
     homepage = "https://github.com/python-poetry/poetry-plugin-export";
     maintainers = [ ];
   };

--- a/pkgs/by-name/po/poetry/plugins/poetry-plugin-up.nix
+++ b/pkgs/by-name/po/poetry/plugins/poetry-plugin-up.nix
@@ -6,6 +6,7 @@
   pytestCheckHook,
   pytest-mock,
   poetry,
+  writableTmpDirAsHomeHook,
 }:
 
 buildPythonPackage rec {
@@ -31,11 +32,14 @@ buildPythonPackage rec {
   nativeCheckInputs = [
     pytestCheckHook
     pytest-mock
+    writableTmpDirAsHomeHook
   ];
 
-  preCheck = ''
-    export HOME=$TMPDIR
-  '';
+  disabledTests = [
+    # https://github.com/MousaZeidBaker/poetry-plugin-up/issues/78
+    "test_command_preserve_wildcard_project"
+    "test_command_with_latest_project"
+  ];
 
   meta = {
     description = "Poetry plugin to simplify package updates";

